gitops/snapshot.go

```diff
@@ -147,6 +150,9 @@ func MarkSnapshotAsPassed(adapterClient client.Client, ctx context.Context, snap
 // MarkSnapshotAsFailed updates the AppStudio Test succeeded condition for the Snapshot to failed.
 // If the patch command fails, an error will be returned.
 func MarkSnapshotAsFailed(adapterClient client.Client, ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot, message string) (*applicationapiv1alpha1.Snapshot, error) {
+	if HaveAppStudioTestsFailed(snapshot) {
```

Comment:
I still think we could make this check in the adapter code before calling  `MarkSnapshotAsFailed()`.

Reply:
Should we add the check for all the places that MarkSnapshotAsFailed() are called in the code ( integrationpipeline_adapter.go / snapshotenvironmentbinding_adapter.go) ?

Reply:
Yup, thanks for making the updates.